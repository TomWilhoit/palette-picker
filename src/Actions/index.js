export const addProjects = (projects) => ({
  type: 'ADD_PROJECTS',
  projects
})

export const addPalettes = (palettes) => ({
  type: 'ADD_PALETTES',
  palettes
})

export const addCurrentProject = (project) => ({
  type: 'ADD_CURRENT_PROJECT',
  project
})