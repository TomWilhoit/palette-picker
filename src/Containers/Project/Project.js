import React, { Component } from 'react';

export class Project extends Component {
  constructor(props) {
    super(props)
    this.state = {
    }
  }


  render() {
    console.log(this.props)
    return(
      <div className='project'>
        <h3>{this.props.name}</h3>
      </div>
    )
  }
}