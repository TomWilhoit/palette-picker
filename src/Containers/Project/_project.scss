.project {
  align-items: center;
  display: flex;
  height: 20%;
  justify-content: space-between;
  margin-top: 10px;
  width: 100%;
}

.project-title {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 100%;
  width: 80%;
  h3 {
    height: 20px;
    padding: 0;
  }
  &:hover {
    cursor: pointer;
    background: $control-btn;
  }
}

.project-delete {
  display: flex;
  justify-content: center;
  width: 20%;
  button {
    @include delete-btn;
    @include small-btn;
  }
}

.active-project {
  background-color: $active;
}