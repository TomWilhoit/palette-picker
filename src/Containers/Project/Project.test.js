import React from "react";
import { Project } from "./Project";
import { mapStateToProps, mapDispatchToProps } from "./Project";
import { shallow } from "enzyme";
import { addProjects, updateCurrentProject, removeProject, removeProjectPalettes } from "../../Actions";

describe("Project", () => {
  let wrapper;
  let mockPalettes = [{ name: "Tom", projectId: 4 }];
  
  beforeEach(() => {
    wrapper = shallow(<Project palettes={mockPalettes} />);
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  })

  it.skip("should have default state", () => {
    expect(wrapper.state()).toEqual({
      id: 0
    });
  })

  it("should call changeCurrentProject when clicked", () => {
    const mockEvent = {
      target: "JIM"
    };
    wrapper.setProps({ updateCurrentProject: jest.fn() });
    jest.spyOn(wrapper.instance(), "changeCurrentProject");
    wrapper.instance().forceUpdate();
    wrapper.find(".project-title").simulate("click", mockEvent);
    expect(wrapper.instance().changeCurrentProject).toHaveBeenCalled();
  })

  it("should call handleDelete when clicked", () => {
    const mockEvent = {
      target: "JIM",
      preventDefault: jest.fn()
    };
    wrapper.setProps({ removeProject: jest.fn(), removeProjectPalettes : jest.fn() });
    jest.spyOn(wrapper.instance(), "handleDelete");
    wrapper.instance().forceUpdate();
    wrapper.find(".project-delete").simulate("click", mockEvent);
    expect(wrapper.instance().handleDelete).toHaveBeenCalled();
  })
  
  it("should mapStateToProps", () => {
    const mockState = {
      currentProject: 4,
      palettes: [{ name: "Mason", projectId: 4 }],
      projects: [{ name: "Tommy" }]
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(mockState);
  })

  it("should map dispatch to props", () => {
    const mockProjects = { name: "Tommy" };
    const mockDispatch = jest.fn();
    const actionToDispatch = addProjects(mockProjects);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.addProjects(mockProjects);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })

  it("should map dispatch to props", () => {
    const mockProject = { name: "Tommy" };
    const mockDispatch = jest.fn();
    const actionToDispatch = updateCurrentProject(mockProject);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.updateCurrentProject(mockProject);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })

  it("should map dispatch to props", () => {
    const mockId = 4;
    const mockDispatch = jest.fn();
    const actionToDispatch = removeProject(mockId);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.removeProject(mockId);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })

  it("should map dispatch to props", () => {
    const mockId = 4;
    const mockDispatch = jest.fn();
    const actionToDispatch = removeProjectPalettes(mockId);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.removeProjectPalettes(mockId);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })
})