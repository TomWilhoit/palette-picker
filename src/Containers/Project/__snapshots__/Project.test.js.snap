// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match the snapshot with all data passed in 1`] = `
<div
  className="project"
>
  <h3
    className="project-title"
    onClick={[Function]}
  />
  <button
    className="project-delete"
    onClick={[Function]}
  >
    <i
      className="fas fa-times"
    />
  </button>
</div>
`;
